### DISCUSSION TITLE
Create Filters from Query Result

### DISCUSSION THREAD

**Discussion by MxFlorin** [rappasoft/laravel-livewire-tables#396] [Q&A] [answered]
Is there a way to create Filters from query result?

Something like :

```
$categories = Categories::all()->map->only('id', 'name');  
$this->filter['category'] = $categories
```

**Comment by rappasoft** [comment, marked as answer]
```php
public function filters(): array
{
    return [
        'categories' => Filter::make('Categories')
            ->select(Categories::all()->map->only('id', 'name')),
    ];
}
```

As long as whatever you send to `select` is `$key => $value`

**Reply by MxFlorin** [reply]
Can I have something like this? 

```
protected $filters = [
    'categories' => '',
]

public function filters(): array
{
    return [
        'categories' => Filter::make('Categories')
            ->select($this->filters['categories']),
    ];
}

public function query():Builder{
    
     $q = Categories:wjere('warehouse', '1') 
     $this->filters['categories'] = $q->pluck('id', 'name');

     return $q;
}
```

What I try to achieve is to update the filters based on query results.

**Reply by rappasoft** [reply]
I don't know if that would work, it should work fine with the first example as I use it on my projects. Then you just apply the filter with the ID that is set in the query string: https://github.com/rappasoft/laravel-livewire-tables/wiki/Creating-filters#2-specifying-how-that-filter-acts-on-the-query
